### PR TITLE
Feature:(issue_1090): Add unwrap for ExitCoder

### LINKED ISSUE #1090: ExitCoder should implement Unwrap
## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

In golang v1.13 the notion of error wrapping was introduced with the following optional method signature on `error` types: `func (*T) Unwrap error`. The `ExitCoder` type is very similar to a wrapped error with addition information (the exit code), so it seems like a good addition.

A long term consideration (for this package's v3) might be to change the implementation of `ExitCoder` to be:

```golang
type exitError struct {
	exitCode int
	err      error
}
```

### PR DESCRIPTION
**Pull request by dearchap**
## What type of PR is this?

_(REQUIRED)_

- feature

## What this PR does / why we need it:

_(REQUIRED)_

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #1090 

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

make test

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -101,20 +101,35 @@ func NewExitError(message interface{}, exitCode int) ExitCoder {
 // by overiding the ExitErrHandler function on an App or the package-global
 // OsExiter function.
 func Exit(message interface{}, exitCode int) ExitCoder {
+	var err error
+
+	switch e := message.(type) {
+	case ErrorFormatter:
```
Comment (Dokiys) [inline errors.go:107]: Is there any difference between this and the default handling?

Comment (dearchap) [reply errors.go:107]: Well ErrorFormatter implies that the error message can be formatted dfferently than a "regular" error.

File: errors.go [hunk 2]
```diff
@@ -101,20 +101,35 @@ func NewExitError(message interface{}, exitCode int) ExitCoder {
...
+		err = fmt.Errorf("%+v", message)
+	case error:
+		err = e
+	default:
+		err = fmt.Errorf("%+v", message)
+	}
```
Comment (Dokiys) [inline errors.go:113]: > Well ErrorFormatter implies that the error message can be formatted dfferently than a "regular" error.

Buf the handling of ErrorFormatter same as default

Comment (dearchap) [reply errors.go:113]: Actually collapsing ErrorFormatter into default doesnt seem to work. I'm going to leave it as it is.

**Pull request merged**